File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> arr) {
    ios::sync_with_stdio(0); cin.tie(0);
    
    vector<int> answer;
    int pre;
    for (int n : arr) {
        if (pre != n) {
            answer.emplace_back(n);
        }
        pre = n;
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int[] arr) {
        List<Integer> list = new ArrayList<>();
        int pre = -1;
        for (int n : arr) {
            if (pre != n) {
                list.add(n);
                pre = n;
            }
        }
        
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); ++i) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}
